### ISSUE TITLE
Backup to remote server

### ISSUE THREAD

**Issue by meganuke20** [zbackup/zbackup#12]
Do you plan this feature?

**Comment by Am1GO** [comment]
No, I'm not going to publish wrapper, I'm going to implement support inside ZBackup itself.

**Comment by thiagoc** [comment]
@Am1GO could you please share your wrapper so we could use in the meantime?

**Comment by Am1GO** [comment]
I'm sorry but no.
It was a bunch of shitty shell scripts so I don't want to make this crap became public.

**Comment by thiagoc** [comment]
Just one question: it's possible to store the repo only on the remote
server? I want to execute my backups from a VirtualBox VM but I don't have
enough disk space to store the repo locally. Currently I'm using Attic that
store everything on the remote end, but I want to start using zbackup.

2015-02-12 11:31 GMT-02:00 Am1GO <notifications@github.com>:

> I'm sorry but no.
> It was a bunch of shitty shell scripts so I don't want to make this crap
> became public.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zbackup/zbackup/issues/12#issuecomment-74070329>.
>



-- 
Thiago Coutinho

"O povo não deveria temer o governo. O governo é quem deveria temer o povo."
V de Vingança

**Comment by korjavin** [comment]
@thiagoc as I understand, you can't.

**Comment by dragonroot** [comment]
You can 1) use sshfs, or 2) move newly created bundles to the remote server after each backup.

**Comment by thiagoc** [comment]
>  2) move newly created bundles to the remote server after each backup.

@dragonroot can you please give me more details about this? I apreciate if you can give a example of this.

**Comment by thiagoc** [comment]
Only the index files are needed to make a new backup?

**Comment by Am1GO** [comment]
@thiagoc 
Yes, it's possible.
For creating a new backup you need only indexes and info, you don't have to store bundles on your client.

**Comment by thiagoc** [comment]
Thanks.

**Comment by mumblez** [comment]
I'm about to try to setup this process, can you guys confirm I have the logic right?

Backup server has zbackup encrypted repo, I want to backup a remote server, so I theoretically:

1. rsync the password file and the info file (of the repo) to the remote server
2. initialise a bare zbackup repo on the remote server
3. copy the info file into the new repo
4. rsync the index directory from backup server to remote server
5. run a backup with the password file into the new repo
6. rsync backup file under backups up to backup server
7. rsync bundle directory up to backup server
8. rsync indexes up to backup server
9. delete info and password file

Can you guys confirm the logic is correct?

**Comment by jonycgn** [comment]
why do you guys want to implement everything in one program? why not just rsync the repository over to another machine, and then expire old bundles with "find -ctime" etc.?

i love that zbackup delegates this, because as a sysadmin, it gives you flexibility to
a) transfer the backup off-site at a different time than creating it
b) use whatever software you like to do the transfer

you don't have to have everything in one executable. think unix, not windows ...

**Comment by jonycgn** [comment]
ok, sorry about my rant, didn't see the comment about not enough disk space ...

**Comment by kevina** [comment]
Hi All,

I wrote a Filesystem in Userspace (FUSE) specifically for zbackup, it uploads bundles as they are created to avoid any space issues. Right now only Google drive is supported but other remotes can be added.  See this post: https://groups.google.com/forum/#!topic/zbackup/t49oZa3LfhE, or my GitHub project: https://github.com/kevina/syncfs if you are interested.

Thanks,
Kevin
